File: lib/Models/appointment_model.dart
class Appointment {
  final int? id;
  final String patientName; // Changed from name
  final String reason;
  final String date;
  final String time;
  final String status;
  final String doctorName;

  Appointment({
    this.id,
    required this.patientName,
    required this.reason,
    required this.date,
    required this.time,
    required this.doctorName,
    this.status = "Pending",
  });

  factory Appointment.fromJson(Map<String, dynamic> json) {
    return Appointment(
      id: json['id'] == null ? null : int.tryParse(json['id'].toString()),
      // Map the DB keys (snake_case) to your Flutter properties
      patientName: json['patient_name'] ?? json['name'] ?? "Unknown",
      doctorName: json['doctor_name'] ?? json['doctorName'] ?? "Unknown",
      reason: json['reason'] ?? "",
      date: json['date'] ?? "",
      time: json['time'] ?? "",
      status: json['status'] ?? "Pending",
    );
  }
}

File: backend/models/appointmentModel.js
const db = require("../config/db");

// GET ALL
exports.getAll = (cb) => {
    // We now pull patient_name and doctor_name directly from the table
    const query = `
        SELECT
            id,
            patient_name,
            doctor_name,
            reason,
            date,
            time,
            status
        FROM appointments
        ORDER BY id DESC
    `;
    db.query(query, cb);
};

// CREATE
exports.create = (data, cb) => {
    const query = `INSERT INTO appointments
                   (user_id, doctor_id, patient_name, doctor_name, reason, date, time, status)
                   VALUES (?, ?, ?, ?, ?, ?, ?, ?)`;

    const values = [
        data.user_id || null,      // Optional ID
        data.doctor_id || null,    // Optional ID
        data.patient_name,         // The String name from Flutter
        data.doctor_name,          // The String name from Flutter
        data.reason,
        data.date,
        data.time,
        data.status || 'Pending'
    ];

    db.query(query, values, cb);
};

// UPDATE STATUS
exports.updateStatus = (id, status, cb) => {
    const query = "UPDATE appointments SET status = ? WHERE id = ?";
    db.query(query, [status, id], cb);
};